cookbooks/vm/recipes/atom.rb

```diff
@@ -1,12 +1,12 @@
 
-atom_version = '1.17.1'
+atom_version = '1.32.2'
```

Comment:
nowadays I would actually install VS Code, but that would be out of scope for this PR anyway..

Reply:
I'm working on that, just not ready for a pull request yet: https://github.com/fasmat/linus-kitchen/commits/customize-installed-applications

Will come with recipes and roles, so you can customize which applications you want with which plugins.

Reply:
Thanks for the heads up! :)

I have only skimmed the changes in the branch, but my first impression was on whether we actually need the configurability / composability there. Personally, I would probably tend towards the opinionated approach and use a fixed runlist with the single `vm::default` recipe. 
Once you start making this configurable, you may run into many more edge cases and discover you have dependencies between the roles, then ordering comes into play, and finally you need the tests to run conditionally depending on whether that role was included or not.
If there are some tools you consider optional, I'd rather have a flag in the json file for this and then conditionally include / exclude that in the default recipe (also saves a lot of role json files).

Just my 5 cents though, and not related to this PR anyway ;)

The VS Code changes Iook promising as well! :)

Reply:
For me it's personal preference. I'd like to have different environments for the development work I do currently. Therefore I'd like to have one environment for golang development, one for .NETCore, one for LaTeX, aso. With only the minimal tools included that I need in that environment.

For testing I yet have to figure out how to only run specific tests based on roles. In another chef project I'm using tags (based on the name of provided run-list). If you know of a better way that would be cool :-)

Reply:
Tags sound like a good approach. Not done that in Chef yet, but something similar in Ansible once.

Ah, now I see what you are up to. The use case to mix and match the tools for the specific purpose / project you need them is of course perfectly valid. Haven't considered it in that context, because linus-kitchen is meant to be a developer VM specifically for "Infrastructure-as-Code" development, thus it is biased / opinionated by nature.

Still, I pretty much like the idea of a general purpose developer VM "framework" where you can then put together and mix and match the toolchains you need. What I believe we need for this would be a central repository for collecting these "toolchain roles" (along with the tests for them), which we could then simply include or reference in the "developer VM" repos. This would also be a huge leap towards better reusability of such roles. I have some ideas on this, would be happy to discuss and exchange if you want.

Reply:
Sure, you know where to find me ;-)